rust_src/src/multibyte.rs

```diff
@@ -310,6 +312,91 @@ impl LispObject {
     pub fn empty_unibyte_string() -> LispStringRef {
         LispStringRef::from(unsafe { empty_unibyte_string })
     }
+
+    // We can excuse not using an option here because extracting the value checks the type
+    // TODO: this is false with the enum model, change this
```

Comment:
Still valid right? The LispObject will either pass `as_string()`, `as_symbol()`, or return `error!()` right?

Reply:
It will, but its behavior has changed slightly. Now it's the during the conversion from `LispObject` to `LispSymbolOrString` the potential error happens, instead of during the conversion from `LispSymbolOrString` to `LispStringRef` or `LispSymbolRef`.

Reply:
Order of errors matters unfortunately. Trace the execution to be sure the Rust code fails where the C code would have. Which Rust function the error happens inside of is not important but when is.

Reply:
As it's currently invoked errors happen when expected, my primary concern is that it doesn't follow the convention of return an option like the other `as_type()` methods, which may lead to misuse. Leaving it as it is would be fine, but it could potentially cause confusion or tech-debt down the line.

Reply:
Ah, yes. Our use of `as_foo()` naming is not really inline with the Rust norm.

https://rust-lang-nursery.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Clippy has a warning about this as well based on the type of `self` but I cannot find it at that moment.